// This file was generated with the Protobuf generator tool.
// Dataset ID    : 10
// Dataset Name  : lymph;
// Dataset URL   : https://www.openml.org/data/v1/download/10/lymph.arff
// Num. Columns  : 19
// Num. Rows     : 148
// Target Feature: class

// Beginning of Description of Dataset:
// **Author**:   
// **Source**: Unknown -   
// **Please cite**:   
// Citation Request:
//     This lymphography domain was obtained from the University Medical Centre,
//     Institute of Oncology, Ljubljana, Yugoslavia.  Thanks go to M. Zwitter and 
//     M. Soklic for providing the data.  Please include this citation if you plan
//     to use this database.
//  1. Title: Lymphography Domain
//  2. Sources: 
//      (a) See Above.
//      (b) Donors: Igor Kononenko, 
//                  University E.Kardelj
//                  Faculty for electrical engineering
//                  Trzaska 25
//                  61000 Ljubljana (tel.: (38)(+61) 265-161
//                  Bojan Cestnik
//                  Jozef Stefan Institute
//                  Jamova 39
//                  61000 Ljubljana
//                  Yugoslavia (tel.: (38)(+61) 
//

syntax = "proto3";

message Empty {
}

message Features {
	string   Lymphatics                     = 1;
	string   Block_of_affere                = 2;
	string   Bl_of_lymph_c                  = 3;
	string   Bl_of_lymph_s                  = 4;
	string   By_pass                        = 5;
	string   Extravasates                   = 6;
	string   Regeneration_of                = 7;
	string   Early_uptake_in                = 8;
	uint32   Lym_nodes_dimin                = 9;
	uint32   Lym_nodes_enlar                = 10;
	string   Changes_in_lym                 = 11;
	string   Defect_in_node                 = 12;
	string   Changes_in_node                = 13;
	string   Changes_in_stru                = 14;
	string   Special_forms                  = 15;
	string   Dislocation_of                 = 16;
	string   Exclusion_of_no                = 17;
	uint32   No_of_nodes_in                 = 18;
	string   Class                          = 19;
}

service get_next_row {
	 rpc get_next_row(Empty) returns(Features);
}

/*
//This code section can be directly used
//at the end of gen_next_row in the server.py file
//for OpenML dataset nr. 10
				response.Lymphatics                     = row[0]
				response.Block_of_affere                = row[1]
				response.Bl_of_lymph_c                  = row[2]
				response.Bl_of_lymph_s                  = row[3]
				response.By_pass                        = row[4]
				response.Extravasates                   = row[5]
				response.Regeneration_of                = row[6]
				response.Early_uptake_in                = row[7]
				response.Lym_nodes_dimin                = numpy.uint32(row[8])
				response.Lym_nodes_enlar                = numpy.uint32(row[9])
				response.Changes_in_lym                 = row[10]
				response.Defect_in_node                 = row[11]
				response.Changes_in_node                = row[12]
				response.Changes_in_stru                = row[13]
				response.Special_forms                  = row[14]
				response.Dislocation_of                 = row[15]
				response.Exclusion_of_no                = row[16]
				response.No_of_nodes_in                 = numpy.uint32(row[17])
				response.Class                          = row[18]
*/

